src/Build.UnitTests/BinaryLogger_Tests.cs

```diff
@@ -99,7 +100,12 @@ public void TestBinaryLoggerRoundtrip(string projectText)
             parallelFromBuild.Parameters = "NOPERFORMANCESUMMARY";
 
             // build and log into binary logger, mock logger, serial and parallel console loggers
-            ObjectModelHelpers.BuildProjectExpectSuccess(projectText, binaryLogger, mockLogFromBuild, serialFromBuild, parallelFromBuild);
+            // no logging on evaluation
+            using (ProjectCollection collection = new())
```

Comment:
this test fails when loggers are also passed to evaluation, whilst I don't see any reason why it should.
Need to investigate the reason.

Reply:
https://github.com/dotnet/msbuild/issues/8414